online-shop/src/app/add-product/add-product.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef } from '@angular/material/dialog';
+import { Supplier } from 'src/supplier';
+import { ProductCategory } from 'src/product-category';
+import { Subscription } from 'rxjs';
+import { ProductCategoryService } from '../services/product-category.service';
+import { SupplierService } from '../services/supplier.service';
+
+@Component({
+    selector: 'app-add-product',
+    templateUrl: './add-product.component.html',
+    styleUrls: ['./add-product.component.scss']
+})
+export class AddProductComponent implements OnInit {
+    addProductForm!: FormGroup;
+    suppliers: Supplier[] = [];
+    productCategories: ProductCategory[] = [];
+    subscription = new Subscription();
+    constructor(
+        private fb: FormBuilder,
+        private dialogRef: MatDialogRef<AddProductComponent>,
+        private categoryService: ProductCategoryService,
+        private supplierService: SupplierService
+    ) {}
+
+    ngOnInit(): void {
+        this.subscription.add(
+            this.categoryService.getAllProductCategories().subscribe((data: ProductCategory[]) => {
```

Comment:
names are very important because a good name will make the code clearer.
In this case, you know that getAllProductCategories() should return productCategories, so you don't have some trivial DATA over there. There will be cases in that you will have a join/takeLatest... on multiple observables and you don't want to have data1, data2 ....